tests/sklearn/test_sklearn_autolog.py

```diff
@@ -0,0 +1,516 @@
+import functools
+import inspect
+from mock import mock
+import warnings
+
+import numpy as np
+import pytest
+import sklearn
+import sklearn.datasets
+
+import mlflow.sklearn
+from mlflow.sklearn.utils import _MIN_SKLEARN_VERSION, _get_arg_names, _truncate_dict
+from mlflow.utils.mlflow_tags import MLFLOW_PARENT_RUN_ID
+from mlflow.utils.autologging_utils import try_mlflow_log
+from mlflow.utils.validation import (
+    MAX_PARAMS_TAGS_PER_BATCH,
+    MAX_PARAM_KEY_LENGTH,
+    MAX_PARAM_VAL_LENGTH,
+)
+
+FIT_FUNC_NAMES = ["fit", "fit_transform", "fit_predict"]
+TRAINING_SCORE = "training_score"
+ESTIMATOR_CLASS = "estimator_class"
+ESTIMATOR_NAME = "estimator_name"
+MODEL_DIR = "model"
+
+
+pytestmark = pytest.mark.large
+
+
+def get_iris():
+    iris = sklearn.datasets.load_iris()
+    return iris.data[:, :2], iris.target
+
+
+def fit_model(model, X, y, fit_func_name):
+    if fit_func_name == "fit":
+        model.fit(X, y)
+
+    if fit_func_name == "fit_transform":
+        model.fit_transform(X, y)
+
+    if fit_func_name == "fit_predict":
+        model.fit_predict(X, y)
+
+    return model
+
+
+def get_run(run_id):
+    return mlflow.tracking.MlflowClient().get_run(run_id)
+
+
+def get_run_data(run_id):
+    client = mlflow.tracking.MlflowClient()
+    data = client.get_run(run_id).data
+    # Ignore tags mlflow logs by default (e.g. "mlflow.user")
+    tags = {k: v for k, v in data.tags.items() if not k.startswith("mlflow.")}
+    artifacts = [f.path for f in client.list_artifacts(run_id)]
+    return data.params, data.metrics, tags, artifacts
+
+
+def load_model_by_run_id(run_id):
+    return mlflow.sklearn.load_model("runs:/{}/{}".format(run_id, MODEL_DIR))
+
+
+def stringify_dict_values(d):
+    return {k: str(v) for k, v in d.items()}
+
+
+def truncate_dict(d):
+    return _truncate_dict(d, MAX_PARAM_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
+
+
+def get_expected_class_tags(model):
+    return {
+        ESTIMATOR_NAME: model.__class__.__name__,
+        ESTIMATOR_CLASS: model.__class__.__module__ + "." + model.__class__.__name__,
+    }
+
+
+def assert_predict_equal(left, right, X):
+    np.testing.assert_array_equal(left.predict(X), right.predict(X))
+
+
+@pytest.fixture(params=FIT_FUNC_NAMES)
+def fit_func_name(request):
+    return request.param
+
+
+@pytest.fixture(autouse=True, scope="function")
+def throw_if_try_mlflow_log_has_emitted_warnings():
+    # autolog contains multiple try_mlflow_log. They may hide unintentional errors (often caused by
+    # incorrect test settings) and allow tests that should not pass to pass (without us noticing).
+    # To prevent that, throw if try_mlflow_log has emitted warnings during test execution.
+    with warnings.catch_warnings(record=True) as ws:
+        yield
+        mlflow_warnings = [w for w in ws if str(w.message).startswith("Logging to MLflow failed")]
+        assert len(mlflow_warnings) == 0
```

Comment:
A fixture to prevent `try_mlflow_log` from allowing tests that should not pass to pass.

Reply:
Relying on a warning message doesn't seem to be a good approach though.